### PR TITLE
[AppService] fix #10664- VNet Integration - Location Check Issue & fix #13257- az webapp up failing when RG needs to be created

### LINKED ISSUE #10664: webapp: VNet Integration - Location Check Issue
**Describe the bug**
While trying to add VNet-integration to the webapp, getting following error: 
`The app and the vnet resources are in different locations.                         Cannot integrate a regional VNET to an app in a different region`

Both application and VNet are in same region. Region used - "East US 2"

**To Reproduce**
1. Use "East US 2" as region for Web app and VNet
2. Also, VNet and Web App were in separate resource groups. Therefore, resource id of the subnet and VNet was used, instead of just name.
/subscriptions/<sub-id>/resourceGroups/<rg-name>/providers/Microsoft.Network/virtualNetworks/<VNet Name>/subnets/servicepoint-subnet

**Expected behavior**
VNet integration should be added without any errors, as both resources are in same region. 

**Environment summary**
CLI Version - 2.0.73
Environment - Visual Studio Code

**Additional context**
I am able to add VNet manually using the portal.

### LINKED ISSUE #13257: az webapp up failing when RG needs to be created
image

When trying to create an app by passing in minimum parameters resource group creation fails since it's missing location, this should be implied from app service defaults.

This works when I pass in location parameter however location parameter should not be required.

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
fix #10664 , #13257 
**Description**  

**Why this PR?** While trying to add VNet-integration to the webapp, getting following error:
The app and the vnet resources are in different locations. Cannot integrate a regional VNET to an app in a different region
**What is changed?**
Removed the region validation up on the confirmation from SME (@panchagnula)
**What is the effect?**
Regional check will not happen whether both the resources are in the same region or not.

**Testing Guide**  

**Steps to reproduce:**
`az group create -n olavs-test-rg -l westeurope`
`az appservice plan create -g olavs-test-rg -n olavs-plan -l westeurope --sku S1`
`az webapp create -g olavs-test-rg -p olavs-plan -n olavs-app`
`az network vnet create -g olavs-test-rg -n olavs-vnet --address-prefixes 10.0.0.0/8`
`az network vnet subnet create -g olavs-test-rg --vnet-name olavs-vnet -n olavs-subnet --address-prefixes 10.0.0.0/16`
`az webapp vnet-integration add -g olavs-test-rg -n olavs-app --vnet olavs-net --subnet olavs-subnet`

**Expected result:** `Command should integrate Vnet to App.`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by Kotasudhakarreddy** [comment]
@panchagnula please review

**Comment by qwordy** [comment]
I highly suggest PR author write a summary for PR. Reviews may not be willing to read the original issue one comment by one comment. A summary not only contains what the problem is, but also contains how you fix it, what's the new behavior, how to test it. Writing a good summary is an important ability to enable others to understand your work.

**Comment by yonzhan** [comment]
add to S169

**Comment by Kotasudhakarreddy** [comment]
> I highly suggest PR author write a summary for PR. Reviews may not be willing to read the original issue one comment by one comment. A summary not only contains what the problem is, but also contains how you fix it, what's the new behavior, how to test it. Writing a good summary is an important ability to enable others to understand your work.

Done! please review.

**Comment by panchagnula** [comment]
@Kotasudhakarreddy I believe we might need to re-record some tests with this change. Can you check?

**Comment by Kotasudhakarreddy** [comment]
> @Kotasudhakarreddy I believe we might need to re-record some tests with this change. Can you check?

@panchagnula -I have re-recorded all the tests from AppService before I did PR. Did you find any missed tests.

**Comment by panchagnula** [comment]
> > @Kotasudhakarreddy I believe we might need to re-record some tests with this change. Can you check?
> 
> @panchagnula -I have re-recorded all the tests from AppService before I did PR. Did you find any missed tests.

The PR has only one file - so I was checking if you are missing adding the recorded files to the PR.

**Comment by Kotasudhakarreddy** [comment]
> @Kotasudhakarreddy I believe we might need to re-record some tests with this change. Can you check?
> 
> @panchagnula -I have re-recorded all the tests from AppService before I did PR. Did you find any missed tests.
> 
> The PR has only one file - so I was checking if you are missing adding the recorded files to the PR.

No Sisira. All the tests were passed after this change.

**Comment by panchagnula** [comment]
Ok - not sure why the build isn't completing though - can you do a pulll & rebase to upstream/dev branch & force push?

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -447,7 +447,7 @@ def load_arguments(self, _):
     with self.argument_context('webapp vnet-integration') as c:
         c.argument('name', arg_type=webapp_name_arg_type, id_part=None)
         c.argument('slot', help="the name of the slot. Default to the productions slot if not specified")
-        c.argument('vnet', help="Vnet name", validator=validate_add_vnet)
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:450]: if this validation isn't used anywhere else remove the validation itself from validators.py

Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:450]: Same function is being used by **functionapp vnet-integration**.

**Comment by panchagnula** [comment]
@Kotasudhakarreddy I don't see the history.rst updated - do we not update that anymore?

**Comment by qwordy** [comment]
> 
> 
> @Kotasudhakarreddy I don't see the history.rst updated - do we not update that anymore?

Right.
https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests

**Pull request merged**